test/SILOptimizer/deinit_barrier.sil

```diff
@@ -0,0 +1,87 @@
+// RUN: %target-sil-opt -compute-side-effects -unit-test-runner %s 2>&1 | %FileCheck %s
```

Comment:
You can keep this test, but please also add tests for the deinit barrier effect in https://github.com/apple/swift/blob/main/test/SILOptimizer/side_effects.sil

Also, please add a test in https://github.com/apple/swift/blob/main/test/SILOptimizer/cross-module-effects.swift to make sure this flag is propagated correctly across modules.

Reply:
Added.